idemp-rest-test/testing/src/tests/charge.test.ts

```diff
@@ -0,0 +1,25 @@
+import { createChargeType, createCharge } from '../utils';
+import { chargeApi } from '../api/charges';
+import { chargeTypeApi } from '../api/chargeTypes';
+
+test('income category is saved and returned', async () => {
+	const INCOME_CATEGORY_CHARGE_TYPE_NAME = 'Default Income Category - DO NOT CHANGE';
+
+	const valueObject = globalThis.__VALUE_OBJECT__;
+	await valueObject.login();
+
+	// get default charge type..
+	const filterJson = '{ "$and": [{"isactive" : {"$eq": true}}, {"c_chargetype": {"$eq" : "' + INCOME_CATEGORY_CHARGE_TYPE_NAME + '" }}]}';
+	const chargeTypes = (await chargeTypeApi.get(valueObject, 0, 1, undefined, filterJson)).results;
+	expect(chargeTypes.length).toBe(1);
+	valueObject.chargeType = chargeTypes[0];
+
+	// create a charge
+	valueObject.stepName = 'Create Charge';
+	await createCharge(valueObject);
+
+	// retrieve charge via rest
+	const charge = (await chargeApi.getByUuid(globalThis.__VALUE_OBJECT__, valueObject.charge!.uuid));
+
+	expect(charge.name).toBe(valueObject.charge!.name);
+});
```

Comment:
```suggestion
});

```

Reply:
what's the reasoning behind adding an extra empty line after the last line?

Reply:
Typically because GitHub shows it as something that should be done.
![image](https://user-images.githubusercontent.com/1461567/187688851-122586cd-41b8-48f3-b273-5a1fd717ff99.png)

I looked up why GH suggests it and found this:
https://stackoverflow.com/questions/5813311/whats-the-significance-of-the-no-newline-at-end-of-file-log

Reply:
very interesting..

Reply:
However, this and other formatting things should be handled automatically (similar to the React App), so I'll add some stuff to handle it.